uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/NavigationMenuHelper.php

```diff
@@ -0,0 +1,236 @@
+  }
+
+  /**
+   * Removes an item from the menu by path
+   *
+   * @param array $menu
+   * @param string $path
+   */
+  public static function remove(&$menu, $path) {
+    $siblings = &self::getSiblingsReference($menu, $path);
+    $itemName = self::getItemNameFromPath($path);
+
+    foreach ($siblings as $index => $sibling) {
+      if ($sibling['attributes']['name'] === $itemName) {
+        unset($siblings[$index]);
+        return;
+      }
+    }
+  }
+
+  /**
+   * Sets the permission on a named menu item
+   *
+   * @param array $menu
+   *   The full menu
+   * @param string $path
+   *   The path to the item, e.g. "Events/Dashboard"
+   * @param string $newPermission
+   *   The new permission for accessing this item
+   */
+  public static function updatePermissionByPath(&$menu, $path, $newPermission) {
+    $item = &self::findMenuItemReferenceByPath($menu, $path);
+
+    if (!$item) {
+      $err = sprintf('Cannot find menu item with path "%s"', $path);
+      throw new \Exception($err);
+    }
+
+    $item['attributes']['permission'] = $newPermission;
+  }
+
+
+  /**
+   * @param array $menu
+   *   The menu array to search in
+   * @param string $path
+   *   The path to the item, e.g. "Events/Dashboard"
+   *
+   * @return array|null
+   *   The menu item if found, or NULL if not
+   */
+  private static function &findMenuItemReferenceByPath(&$menu, $path) {
+    $path = explode('/', $path);
+    $submenu = &$menu;
+
+    while (count($path) > 1) {
+      $parentName = array_shift($path);
+      $submenu = &self::findMenuItemReferenceByName($submenu, $parentName);
+      if (!isset($submenu['child'])) {
+        $null = NULL; // must return reference to variable
+
+        return $null;
+      }
+      $submenu = &$submenu['child'];
+    }
+
+    $itemName = array_shift($path);
+    $item = &self::findMenuItemReferenceByName($submenu, $itemName);
+
+    return $item;
+  }
+
+  /**
+   * Search for a menu item by its name. Returns a reference to the item.
+   *
+   * @param array $menu
+   *   The menu to search, can be a submenu of the full structure
+   * @param string $name
+   *   The name of the item to search for
+   *
+   * @return array|null
+   *   The referenced menu item, if found, or null otherwise
+   */
+  private static function &findMenuItemReferenceByName(&$menu, $name) {
+    foreach ($menu as &$item) {
+      if ($item['attributes']['name'] === $name) {
+        return $item;
+      }
+    }
+
+    // must return reference to variable
+    $item = NULL;
+
+    return $item;
+  }
+
+  /**
+   * Returns an array containing all sibling items on the same menu level.
+   * This includes the target item.
+   *
+   * @param array $menu
+   *   The full menu structure
+   * @param string $path
+   *   The path of the menu item whose siblings we want
+   *
+   * @return array
+   */
+  private static function &getSiblingsReference(&$menu, $path) {
+    $finalForwardSlashPos = strrpos($path, '/');
```

Comment:
Won't this return the position of the first forward slash?

Reply:
`strrpos` !== `strpos`

Reply:
oh, didn't catch the two rr's.